packages/blade/src/components/Collapsible/_decisions/decisions.md

```diff
@@ -0,0 +1,186 @@
+
+Built-in trigger for `Collapsible`, provides a subset of `Button` [properties](../../Button/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `variant`
+- `size`
+- `icon`
+- `iconPosition`
+- `isDisabled`
+- `testID`
+- `accessibilityLabel`
+
+Doesn't support ❌
+
+- `isFullWidth` - can visually break UI
+- `onClick`, `isLoading` - using `onClick` will break built-in interactions (unless we provide additional hooks), `onExpandChange` should already provide a way for users to listen-in on `Collapsible` state. This might have usage for async with `isLoading` in the future but for now not exposing the prop
+- `type` - will be `button` always
+
+### CollapsibleLink
+
+Built-in trigger for `Collapsible`, provides a subset of `Link` [properties](../../Link/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `isDisabled`
+- `size`
+- `accessibilityLabel`
+- `testID`
+
+Doesn't support ❌
+
+- `icon` - renders a chevron icon always
+- `iconPosition` - will be `right` always for the chevron icon (which animates)
+- `htmlTitle`
+- `variant` - will always be `button`
+- `onClick`
+- `href`, `target`, `rel` - will always be a button
+
+### CollapsibleBody
+
+Slot for rendering `Collapsible` content
+
+| Prop     | Type  | Default     | Description                      | Required |
+| -------- | ----- | ----------- | -------------------------------- | -------- |
+| children | `JSX` | `undefined` | Slot, renders any custom content | ✅       |
+
+## a11y
+
+Collapsible shares a11y concepts with accordion.
+
+Web:
+
+Follows the standard practices (aria expanded, controls, etc.) [ARIA guide - disclosure](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/), [ARIA guide - accordion](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/) [example](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/#rps_label):
+
+- <kbd>Enter</kbd> or <kbd>Space</kbd> can toggle collapsible
+- Collapsible is focussable
+- `aria-expanded` and `aria-controls` in the button trigger
+- `aria-labelledby` and `role: region` in the content
+
+Native:
+
+- Some related properties from web will be applicable to native as well ([a11y guide](https://reactnative.dev/docs/accessibility))
+
+## Usage
+
+### Controlled state
+
+```tsx
+const App = () => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const onExpandChange = ({ isExpanded }) => setIsExpanded(isExpanded);
+
+  return (
+    <Collapsible isExpanded={isExpanded} onExpandChange={onExpandChange}>
+      <CollapsibleLink>Answer to life, universe and everything</CollapsibleLink>
+      <CollapsibleBody>
+        <Text>42</Text>
+      </CollapsibleBody>
+    </Collapsible>
+  );
+};
+```
+
+## Alternatives
+
+- Considered a more flexible API of using anything as a collapsible trigger instead of restricted wrappers over `Button` and `Link`. However, this can lead to inconsistencies and a11y issues _(because users will then have to handle all the a11y props and focus management)_.
```

Comment:
Sorry if I am not understanding it correctly, but by this do you mean that you don't encourage having custom triggers for Collapsible but the API gives button and link separately, which means we can easily skip having the button or link triggers and can make a fully controlled Collapsible component just using CollapsbileBody

Reply:
also if you use CollapsibleBody without Collapsible it'll likely throw error, right @divyanshu013 ?

Reply:
Right, internally it uses a Collapsible context and would throw